contracts/zero-ex/contracts/src/features/native_orders/NativeOrdersSettlement.sol

```diff
@@ -514,6 +559,87 @@ abstract contract NativeOrdersSettlement is
         );
     }
 
+    /// @dev Fill an taker signed rfq order. Private variant. Does not refund protocol fees.
+    /// @param order The Taker Signed RFQ order.
+    /// @param makerSignature The order signature from the maker.
+    /// @param takerSignature The order signature from the taker.
+    /// @return results Results of the fill.
+    function _fillTakerSignedRfqOrderPrivate(
+        LibNativeOrder.TakerSignedRfqOrder memory order,
+        LibSignature.Signature memory makerSignature,
+        LibSignature.Signature memory takerSignature
+    )
+        private
+        returns (FillNativeOrderResults memory results)
+    {
+        LibNativeOrder.TakerSignedOrderInfo memory takerSignedOrderInfo = getTakerSignedRfqOrderInfo(order);
+
+        // Must be fillable.
+        if (takerSignedOrderInfo.status != LibNativeOrder.OrderStatus.FILLABLE) {
+            LibNativeOrdersRichErrors.OrderNotFillableError(
+                takerSignedOrderInfo.orderHash,
+                uint8(takerSignedOrderInfo.status)
+            ).rrevert();
+        }
+
+        {
+            LibNativeOrdersStorage.Storage storage stor =
+                LibNativeOrdersStorage.getStorage();
+
+            // Must be fillable by the tx.origin.
+            if (order.txOrigin != tx.origin && !stor.originRegistry[order.txOrigin][tx.origin]) {
```

Comment:
Have we evaluated whether this could pose a risk if the origin registries for RFQ and RFQM are the same? It feels pretty low stakes but just wondering if anyone else had other thoughts.

Reply:
I was thinking it would be fine.

I can't think of a case where you'd specify an address with registered tx origins and not be okay with it for one order type and not the others. In either case you're giving power of tx submission to the entity behind the specified tx origin who may or may not have delegated other senders.